cmake/kokkos_arch.cmake

```diff
@@ -450,11 +450,13 @@ IF (KOKKOS_ARCH_POWER9)
 ENDIF()
 
 IF (KOKKOS_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE)
-  COMPILER_SPECIFIC_FLAGS(
-    Clang  -fcuda-rdc
-    NVIDIA --relocatable-device-code=true
-    NVHPC -gpu=rdc
-  )
+  IF (NOT KOKKOS_COMPILE_LANGUAGE STREQUAL CUDA)
```

Comment:
This does not seem right
Assume RDC is OFF but we are using NVHPC L462 we'd still be adding the `-gpu=nordc` flag

Reply:
I agree. This if condition should guard from setting any flags.